### PR TITLE
avoid potential segfault that may be caused by neural_net_mock

### PR DESCRIPTION
**Pull request by Jarvi-Izana**
learned the mock mechanism and put some change. Hopefully, it will help.

__MY OVERVIEW__
When the mock model backend object's `create_xxx` is called, instead of returning `nullptr`, just throw a `NotImplemented` exception is a better design.

__MY REVIEW SUGGESTION__
* `neural_net_mocks.hpp` contains all relevant changes to prevent a segfault.
* `test_drawing_classifier.cxx` contains some minor changes that @shantanuchhabra is required to review and others are optional but welcomed. I missed my code review for this section, and now it's time to remedy. Well, Unix style relative include should be avoided. If I have more bandwidth, I will spend time addressing that.

__MY SUMMARY__
* [Y] include unittest? [Y/N]
* [Y] pass unittest? [Y/N]

### CODE REVIEW

File: test/unity/toolkits/drawing_classifier/test_drawing_classifier.cxx [hunk 1]
```diff
@@ -199,26 +200,27 @@ BOOST_AUTO_TEST_CASE(test_drawing_classifier_init_training) {
...
-                                 w[i] = static_cast<float>(i);
-                               }
-                             });
+                             model_spec::padding_type::SAME, weight_init_fn);
     return nn_spec;
   });
 
+  /**
+   * TODO: const float_array_map& weights, const float_array_map&
+   * config. Until the nn_spec in C++ isn't ready, do not pass in any
```
Comment (shantanuchhabra) [inline test/unity/toolkits/drawing_classifier/test_drawing_classifier.cxx]: Having `config` be in a separate line makes this comment slightly less readable, I think.

Comment (Jarvi-Izana) [reply test/unity/toolkits/drawing_classifier/test_drawing_classifier.cxx]: well, you will remove this later after you finish the TODO?.

Comment (shantanuchhabra) [reply test/unity/toolkits/drawing_classifier/test_drawing_classifier.cxx]: Yes, this TODO is partially updated in #2482

File: test/unity/toolkits/neural_net/neural_net_mocks.hpp [hunk 2]
```diff
@@ -163,17 +183,18 @@ class mock_compute_context : public compute_context {
   std::unique_ptr<model_backend> create_activity_classifier(
       int n, int c_in, int h_in, int w_in, int c_out, int h_out, int w_out,
       const float_array_map& config, const float_array_map& weights) override {
-    return nullptr;
+    throw std::runtime_error("create_activity_classifier not implemented");
```
Comment (Jarvi-Izana) [inline test/unity/toolkits/neural_net/neural_net_mocks.hpp]: better to throw if it's not implemented.

File: test/unity/toolkits/neural_net/neural_net_mocks.hpp [hunk 3]
```diff
@@ -163,17 +183,18 @@ class mock_compute_context : public compute_context {
...
   }
 
   std::unique_ptr<model_backend> create_style_transfer(
       const float_array_map& config, const float_array_map& weights) override {
-    return nullptr;
```
Comment (Jarvi-Izana) [inline test/unity/toolkits/neural_net/neural_net_mocks.hpp]: same here.

**Pull request merged**